packages/rest-crud/src/crud-rest-builder.plugin.ts

```diff
@@ -0,0 +1,123 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/rest-crud
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {bind, ControllerClass, inject} from '@loopback/core';
+import {
+  asModelApiBuilder,
+  ModelApiBuilder,
+  ModelApiConfig,
+} from '@loopback/model-api-builder';
+import {
+  ApplicationWithRepositories,
+  Class,
+  Entity,
+  EntityCrudRepository,
+} from '@loopback/repository';
+import {Model} from '@loopback/rest';
+import debugFactory from 'debug';
+import {defineCrudRepositoryClass, defineCrudRestController} from '.';
+
+const debug = debugFactory('loopback:boot:crud-rest');
+
+export interface ModelCrudRestApiConfig extends ModelApiConfig {
+  // E.g. '/products'
+  basePath: string;
+}
+
+@bind(asModelApiBuilder)
+export class CrudRestApiBuilder implements ModelApiBuilder {
+  readonly pattern: string = 'CrudRest';
+
+  build(
+    application: ApplicationWithRepositories,
+    modelClass: typeof Model & {prototype: Model},
+    cfg: ModelApiConfig,
+  ): Promise<void> {
+    const modelName = modelClass.name;
+    const config = cfg as ModelCrudRestApiConfig;
+    if (!config.basePath) {
+      throw new Error(
+        `Missing required field "basePath" in configuration for model ${modelName}.`,
+      );
+    }
+
+    if (!(modelClass.prototype instanceof Entity)) {
+      throw new Error(
+        `CrudRestController requires an Entity, Models are not supported. (Model name: ${modelName})`,
+      );
+    }
+    const entityClass = modelClass as typeof Entity & {prototype: Entity};
+
+    let repoClassName = entityClass.name + 'Repository';
+
+    try {
+      application.getBinding('repositories.' + repoClassName);
+      debug(
+        'Using repository class',
+        repoClassName,
+        ', as it is already bound to application',
+      );
+    } catch {
+      // repository class does not exist
```

Comment:
We should use `application.isBound('repositories.' + repoClassName)` to test instead of `try-catch`.